pyuvdata/utils.pyx

```diff
@@ -19,21 +21,35 @@ from libc.math cimport atan2, cos, sin, sqrt
 # We may be able to just remove this then.
 numpy.import_array()
 
+cdef class Ellipsoid:
+  cdef readonly numpy.float64_t gps_a, gps_b, e_squared, e_prime_squared, b_div_a2
+
+  @cython.cdivision
+  def __init__(self, numpy.float64_t gps_a, numpy.float64_t gps_b):
+    self.gps_a = gps_a
+    self.gps_b = gps_b
+    self.b_div_a2 = (self.gps_b / self.gps_a)**2
+    self.e_squared = (1 - self.b_div_a2)
+    self.e_prime_squared = (self.b_div_a2**-1 - 1)
+
+
+# A python interface for different celestial bodies
+class Body(enum.Enum):
+  Earth = Ellipsoid(6378137, 6356752.31424518)
+  # moon data taken from https://nssdc.gsfc.nasa.gov/planetary/factsheet/moonfact.html
+  # with radius from spice_utils
+  Moon = Ellipsoid(1737.1e3, 1737.1e3 * (1 - 0.0012))
+
+
+# expose up to python
 # in order to not have circular dependencies
 # define transformation parameters here
 # parameters for transforming between xyz & lat/lon/alt
-cdef numpy.float64_t _gps_a = 6378137
-cdef numpy.float64_t _gps_b = 6356752.31424518
-cdef numpy.float64_t _e2 = 6.69437999014e-3
-cdef numpy.float64_t _ep2 = 6.73949674228e-3
-# this one is useful in the xyz from lla calculation
-cdef numpy.float64_t b_div_a2 = (_gps_b / _gps_a)**2
-
-# expose up to python
-gps_a = _gps_a
-gps_b = _gps_b
-e_squared = _e2
-e_prime_squared = _ep2
+# keep for consistent API though these really shouldn't be used anymore
+gps_a = Body.Earth.value.gps_a
+gps_b =  Body.Earth.value.gps_b
+e_squared =  Body.Earth.value.e_squared
+e_prime_squared =  Body.Earth.value.e_prime_squared
```

Comment:
Is anything in pyuvdata using these? Or is this for other packages?

Reply:
This is the only file I see these constants being used, just didn't want to remove it because inevitably someone somewhere is using it

Reply:
😆 One option would be to remove them in v3.0 where we're allowed to make breaking API changes...

Reply:
I'm not opposed!